components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/CacheBackedRoleDAO.java

```diff
@@ -0,0 +1,365 @@
+    }
+
+    @Override
+    public void updateIdpGroupListOfRole(String roleId, List<IdpGroup> newGroupList,
+                                         List<IdpGroup> deletedGroupList, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        roleDAO.updateIdpGroupListOfRole(roleId, newGroupList, deletedGroupList, tenantDomain);
+    }
+
+    @Override
+    public void deleteRole(String roleId, String tenantDomain) throws IdentityRoleManagementException {
+
+        Role role = roleDAO.getRole(roleId, tenantDomain);
+        RoleNameCacheKey cacheKey = new RoleNameCacheKey(role.getName(), role.getAudience(), role.getAudienceId());
+        roleCacheByName.clearCacheEntry(cacheKey, tenantDomain);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Cache cleared for role: " + role.getName());
+        }
+        roleDAO.deleteRole(roleId, tenantDomain);
+    }
+
+    @Override
+    public void updateRoleName(String roleId, String newRoleName, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        Role role = roleDAO.getRole(roleId, tenantDomain);
+        RoleNameCacheKey cacheKey = new RoleNameCacheKey(role.getName(), role.getAudience(), role.getAudienceId());
+        roleCacheByName.clearCacheEntry(cacheKey, tenantDomain);
+        roleDAO.updateRoleName(roleId, newRoleName, tenantDomain);
+    }
+
+    @Override
+    public List<UserBasicInfo> getUserListOfRole(String roleId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return roleDAO.getUserListOfRole(roleId, tenantDomain);
+    }
+
+    @Override
+    public void updateGroupListOfRole(String roleId, List<String> newGroupIDList, List<String> deletedGroupIDList,
+                                      String tenantDomain) throws IdentityRoleManagementException {
+
+        roleDAO.updateGroupListOfRole(roleId, newGroupIDList, deletedGroupIDList, tenantDomain);
+    }
+
+    @Override
+    public List<GroupBasicInfo> getGroupListOfRole(String roleId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        return roleDAO.getGroupListOfRole(roleId, tenantDomain);
+    }
+
+    @Override
+    public void updateUserListOfRole(String roleId, List<String> newUserIDList, List<String> deletedUserIDList,
+                                     String tenantDomain) throws IdentityRoleManagementException {
+
+        roleDAO.updateUserListOfRole(roleId, newUserIDList, deletedUserIDList, tenantDomain);
+    }
+
+    @Override
+    public String getRoleNameByID(String roleId, String tenantDomain) throws IdentityRoleManagementException {
+
+        return roleDAO.getRoleNameByID(roleId, tenantDomain);
+    }
+
+    @Override
+    public String getRoleIdByName(String roleName, String audience, String audienceId, String tenantDomain)
+            throws IdentityRoleManagementException {
+
+        RoleNameCacheKey cacheKey = new RoleNameCacheKey(roleName, audience, audienceId);
```

Comment:
Let's check the usages of this cache. If it is only used for everyone `role` let's have it only for that.

Reply:
Addressed